.github/workflows/test.yml

```diff
@@ -12,34 +12,77 @@ jobs:
     name: Run unit tests
     steps:
       - name: Checkout repository
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           ref: ${{ github.ref }}
-      - name: Setup .npmrc file for NPM registry
-        uses: actions/setup-node@v2
+      - name: Setup node
+        uses: actions/setup-node@v3
         with:
           node-version: '16'
-          registry-url: 'https://registry.npmjs.org'
+          cache: 'yarn'
       - name: Install dependencies
-        run: yarn
+        run: yarn install --frozen-lockfile
       - name: Check types
         run: yarn typecheck
       - name: Lint
         run: yarn lint
       - name: Unit tests
         run: yarn test:ci
+      - name: Build packages
+        run: yarn build
+      - name: Pack packages
+        run: yarn package:ci
+      - name: Upload packaged artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: built-packages
+          path: |
+            packages/create-plugin/create-plugin.tgz
+            packages/sign-plugin/sign-plugin.tgz
+  commands:
+    needs: test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
```

Comment:
Should we extend it with `macos-latest` at least? (I am not sure `windows-latest` would work out of the box as well, but maybe it should?)